src/pages/index.js

```diff
@@ -15,86 +16,215 @@ import {
+  addCardPopup.close();
+  addCardFormElement.reset();
+  addCardValidator.disableSubmitButton();
+  api
+    .createCard({ name, link })
+    .then((res) => {
+      const card = createCard(res);
+      section.addItem(card);
+    })
+    .finally(() => {
+      addCardPopup.setLoading(false);
+    });
+}
 
-//preview Image Modal
+//preview card image functionality
 const previewImageModal = new PopupWithImage("#preview-image-modal");
 previewImageModal.setEventListeners();
 
 function handleImagePreview(card) {
   previewImageModal.open(card);
 }
 
-//edit profile functionality
+//delete card functionality
+function handleDeleteCard(cardId, cardElement) {
+  deleteModalConfirmation.open(cardId, cardElement);
+  console.log(cardId, cardElement);
+}
+
+const deleteModalConfirmation = new PopupConfirmation(
+  "#modal-confirm",
+  (cardId, cardElement) => {
+    return api.deleteCard(cardId).then(() => {
+      cardElement.remove();
+    });
+  }
+);
+
+//USER INFO FUNCTIONS
+
+//get user info
 const userInfo = new UserInfo({
   name: ".profile__title",
   description: ".profile__description",
+  avatarSelector: ".profile__image",
 });
 
+api
+  .getUserInfo()
+  .then((userData) => {
+    userInfo.setUserInfo({
+      name: userData.name,
+      description: userData.about,
+      avatar: userData.avatar,
+    });
+  })
+  .catch((err) => console.error(err));
+
+//edit profile functionality
 const editProfilePopup = new PopupWithForm(
   "#profile-edit-modal",
   handleProfileEditSubmit
 );
 editProfilePopup.setEventListeners();
 
+function handleProfileEditSubmit(data) {
+  api
+    .editUserInfo({
+      name: data.title,
+      about: data.description,
+    })
+    .then((userData) => {
+      userInfo.setUserInfo({
+        name: userData.name,
+        description: userData.about,
+        avatar: userData.avatar,
+      });
+    })
+    .catch((err) => console.error(`Failed to update users info: ${err}`))
+    .finally(() => {
+      editProfilePopup.setLoading(false);
+    });
+
+  editProfilePopup.close();
+}
+
 profileEditButton.addEventListener("click", () => {
   const currentUserInfo = userInfo.getUserInfo();
   profileTitleInput.value = currentUserInfo.name;
   profileDescriptionInput.value = currentUserInfo.description;
   editProfilePopup.open();
 });
 
-function handleProfileEditSubmit(formData) {
-  userInfo.setUserInfo({
-    name: formData.title,
-    description: formData.description,
-  });
-  editProfilePopup.close();
-}
-
-//add card functionality
-const addCardPopup = new PopupWithForm(
-  "#add-card-modal",
-  handleAddCardFormSubmit
+// edit user avatar functionality
+const editAvatarModal = new PopupWithForm(
+  "#edit-avatar-modal",
+  handleAvatarEditFormSubmit
 );
-addCardPopup.setEventListeners();
+editAvatarModal.setEventListeners();
 
-addNewCardButton.addEventListener("click", () => {
-  addCardPopup.open();
+editAvatarButton.addEventListener("click", () => {
+  avatarFormValidator.resetValidation();
```

Comment:
`resetValidation` is for clearing possible validation errors when you open the **profile** form, because you insert **valid** data into its inputs.  


You don't need it here


Please, call `disableSubmitButton` in line 225 in `then`

Reply:
* You haven't corrected another remark https://github.com/gerard415/se_project_aroundtheus/pull/1/files#r1777545055